zerver/tests/test_decorators.py

```diff
@@ -327,25 +332,25 @@ def my_webhook_raises_exception_unsupported_event(
         # Test when content_type is application/json but request.body
         # is not valid JSON; invalid JSON should be logged and the
         # exception raised in the webhook function should be re-raised
-        with mock.patch('zerver.decorator.webhook_logger.exception') as mock_exception:
+        with self.assertLogs('zulip.zerver.webhooks', level='ERROR') as log:
             with self.assertRaisesRegex(Exception, "raised by webhook function"):
                 request.body = "invalidjson"
                 request.content_type = 'application/json'
                 request.META['HTTP_X_CUSTOM_HEADER'] = 'custom_value'
                 my_webhook_raises_exception(request)
 
-        mock_exception.assert_called_with("raised by webhook function", stack_info=True)
+        self.assertIn(self.logger_output("raised by webhook function\n", 'error', 'webhooks'), log.output[0])
```

Comment:
Does this need to be `assertIn`? Our other tests use `assertEqual(log.output, [self.logger_output(…)])`.

Reply:
We initially tried to use assertEqual, we found that the message we were looking for ("raised by webhook function") was only a component of a longer message that was provided by the log.output, so it would never be equal.